### PR TITLE
RDoc-3009 [Node.js] Subscriptions > Advanced > Maintenance operations 

### PR DESCRIPTION
**Pull request by Danielle9897**
**Related issue**
https://issues.hibernatingrhinos.com/issue/RDoc-3009/Node.js-Client-API-Data-subscriptions-Advanced-Maintenance-operations-Replace-C-samples

---

**Notes for this PR:**
There are still fixes to be applied to this article for all languages - to be done in a separate dedicated issue:
https://issues.hibernatingrhinos.com/issue/RDoc-3055/Client-API-Data-subscriptions-Advanced-Maintenance-operations-Fix-article

---

**Node.js**: @ml054 
* Node.js files to review:
```
Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown
Documentation/5.4/Samples/nodejs/client-api/dataSubscriptions/advanced/maintenance.js
```

### CODE REVIEW

File: Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+{CODE:nodejs get_state_syntax@client-api\dataSubscriptions\advanced\maintenance.js /}
+
+{INFO: }
+
+##### SubscriptionState
+
+| Member                                    | Type         | Description                                                                                                                                                 |
+|-------------------------------------------|--------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------| 
+| **query**                                 | `string`     | Subscription's RQL like query.                                                                                                                              |
+| **lastBatchAckTime**                      | `DateTime?`  | Last time a batch processing progress was acknowledged.                                                                                                     |
```
Comment (ml054) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown:91]: Date?

File: Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+| **nodeTag**                               | `string`     | Processing server's node tag.                                                                                                                               |
+| **mentorNode**                            | `string`     | The mentor node that was manually set.                                                                                                                      |
+| **subscriptionName**                      | `string`     | The subscription's name, which is also its unique identifier.                                                                                               |
+| **subscriptionId**                        | `long`       | Subscription's internal identifier (cluster's operation etag during subscription creation).                                                                 |
```
Comment (ml054) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown:95]: number?

File: Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+| **changeVectorForNextBatchStartingPoint** | `string`     | The Change Vector from which the subscription will begin sending documents.<br>This value is updated on batch acknowledgement and can also be set manually. |
+| **disabled**                              | `bool`       | If `true`, subscription will not allow workers to connect.                                                                                                  |
+| **lastClientConnectionTime**              | `DateTime?`  | Time when last client was connected (value sustained after disconnection).                                                                                  |
```
Comment (ml054) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown:98]: please review entire table

File: Documentation/5.4/Samples/nodejs/client-api/dataSubscriptions/advanced/maintenance.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+async function revisionsSubscription() {
+
+    {
+        //region delete
+        await documentStore.subscriptions.delete("subscriptionNameToDelete");
+        //endregion
+    }    
+    {
+        //region disable
+        await documentStore.subscriptions.disable("subscriptionNameToDidable");
```
Comment (ml054) [inline Documentation/5.4/Samples/nodejs/client-api/dataSubscriptions/advanced/maintenance.js:16]: typo: Didable

File: Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown (hunk 1 above)
Comment (Danielle9897) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown]: I modified to a `string` as I see here:
https://github.com/ravendb/ravendb-nodejs-client/blob/b82a20977fb56f0928e1eb918042b9e808cbd0ee/src/Documents/Subscriptions/SubscriptionState.ts#L4

File: Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown (hunk 2 above)
Comment (Danielle9897) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown]: right...

File: Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown (hunk 3 above)
Comment (Danielle9897) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/data-subscriptions/advanced-topics/maintenance-operations.js.markdown]: done

File: Documentation/5.4/Samples/nodejs/client-api/dataSubscriptions/advanced/maintenance.js (hunk 4 above)
Comment (Danielle9897) [reply Documentation/5.4/Samples/nodejs/client-api/dataSubscriptions/advanced/maintenance.js]: done

**Review by ml054 — APPROVED** [review]

**Pull request merged**